tools/bin/make/airbyte_ci_install.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Base URL for downloading airbyte-ci
+RELEASE_URL=${RELEASE_URL:-"https://connectors.airbyte.com/files/airbyte-ci/releases"}
+
+# Determine the operating system and download the binary
+OS=$(uname)
+VERSION=${1:-latest}
+
+
+if [ "$OS" = "Linux" ]; then
+    URL="${RELEASE_URL}/ubuntu/${VERSION}/airbyte-ci"
+    echo "Linux based system detected. Downloading from $URL"
+elif [ "$OS" = "Darwin" ]; then
+    URL="${RELEASE_URL}/macos/${VERSION}/airbyte-ci"
+    echo "macOS based system detected. Downloading from $URL"
+else
+    echo "Unsupported operating system"
+    exit 1
+fi
+
+# Create the directory if it does not exist
+mkdir -p ~/.local/bin
+
+# Remove the old binary if it exists
+rm -f ~/.local/bin/airbyte-ci
```

Comment:
Consider `curl`ing the new binary first, before removing the old one. The download may fail and then the user is up the proverbial creek.

Reply:
Yeah fair. I got lazy and ignored doing a tmp folder

Reply:
If you create a temp folder then you also need to add an exit trap to clean it up and those are easy to not get right. Consider doing a `curl -o` instead.

Reply:
I did have -o set. but it was failing to overwrite and kept leaving the dev sym link.

Ill look into it a bit more

Reply:
Did some research, the main pain is that curls download will follow the symbolic link.

To get it not to, or to delete the symbolic link with a fallback that recreates it is a bit of fiddly code
https://superuser.com/questions/303559/replace-symbolic-links-with-files

Given that once the binary is widely used we will be removing the `airbyte-ci` python entry point in favour of `airbyte-ci-dev` this symlink issue should go away.

so for not I opted to only delete the destination path when its a symlink.

Worst case scenario they run clean and install again